src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java

```diff
@@ -295,13 +335,25 @@ public void onTraversalStart() {
         final LinkedHashMap<String, String> unaccountedForOverrideAnnotations = getUnaccountedForAnnotations( dataSourceFactories, annotationOverridesMap );
 
         // Set up our output renderer:
-        // TODO: in the future this should be encapsulated into a factory for output renderers based on an input argument.
-        outputRenderer = new VcfOutputRenderer(getHeaderForVariants(),
-                                               createVCFWriter(outputFile),
-                                               dataSourceFactories,
-                                               unaccountedForDefaultAnnotations,
-                                               unaccountedForOverrideAnnotations);
-
+        switch (outputFormatType) {
+            case MAF:
+                outputRenderer = new MafOutputRenderer(outputFile.toPath(),
+                        dataSourceFactories,
+                        unaccountedForDefaultAnnotations,
+                        unaccountedForOverrideAnnotations,
+                        getHeaderForVariants(),
+                        getDefaultToolVCFHeaderLines().stream().map(x -> x.toString()).collect(Collectors.toCollection(LinkedHashSet::new)));
+                break;
+            // Default to VCF output:
+            default:
+                outputRenderer = new VcfOutputRenderer(getHeaderForVariants(),
+                        createVCFWriter(outputFile),
+                        dataSourceFactories,
+                        unaccountedForDefaultAnnotations,
+                        unaccountedForOverrideAnnotations,
+                        getDefaultToolVCFHeaderLines());
+                break;
+        }
```

Comment:
Another (better, IMO) way to avoid this issue is to put the logic in the enum. This code becomes 
```
outputRenderer  = outputFormatType.createRenderer(outputFile, dataSourceFactories, unaccountedForDefaultAnnotations, unaccountedForOverrideAnnotations, getHeaderForVariants(), getDefaultToolVCFHeaderLines();
```
The enum code would look like
```
    enum OutputFormatType {
        MAF {
            OutputRenderer createRenderer(...) {
                return new MafOutputRenderer(...);
            }
        },
        VCF {
            OutputRenderer createRenderer(...) {
                return VcfOutputRenderer(...);
            }
        };
        abstract OutputRenderer createRenderer();
    }
```
This is better as it doesn't require `default` and it requires that any future change to `OutputFormatType` requires adding a new `createRenderer()` method.

Reply:
Sounds good.

Fixed!

Reply:
@pshapiro4broad Good suggestion.  I'll change it to do this.